client/service-container-chain/src/monitor.rs

```diff
@@ -59,10 +60,10 @@ pub struct SpawnedContainer {
     /// Used to check the reference count, if it's 0 it means the database has been closed
     pub backend: std::sync::Weak<ContainerChainBackend>,
     /// Used to check the reference count, if it's 0 it means that the client has been closed.
-    pub client: std::sync::Weak<ContainerChainClient>,
```

Comment:
This is unfortunate, this module only cares about the refcount so it shouldn't need this generic param everywhere... But I believe we would need unsafe code to remove it.

Reply:
Actually, would it work if you set the type to `dyn Any`? Try initializing it like this

```
client.clone() as Arc<dyn Any + Send + Sync>;
```